### PR TITLE
Javascript: New Express Js Fathead

### LINKED ISSUE #811: JS: New Express.js Fathead
# Create a new Express Fathead Instant Answer
Help us make DuckDuckGo the best search engine for programmers!

### What do I need to know?
You'll need to know how to code in **Perl**, **Python**, **Ruby**, or **JavaScript**.
fathead languages

### What am I doing?
You will write a script that scrapes or downloads the data source below, and generates an **output.txt** file containing the parsed documentation. You can learn more about Fatheads and the `output.txt` syntax [**here**](https://docs.duckduckhack.com/resources/fathead-overview.html).

**Data source**: http://expressjs.com/en/4x/api.html

**Instant Answer Page**: *to be created by developer*

### What is the Goal?
As part of our [Programming Mission](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53), we're aiming to reach 100% Instant Answer (IA) coverage for searches related to programming languages by creating new Instant Answers, and improving existing ones.

Here are some Fathead examples:

- Ruby Docs
    - [Code](https://github.com/duckduckgo/zeroclickinfo-fathead/tree/master/lib/fathead/ruby) | [Example Query](https://duckduckgo.com/?q=array+bsearch&ia=about)
- MDN CSS
    - [Code](https://github.com/duckduckgo/zeroclickinfo-fathead/tree/master/lib/fathead/mdn_css) | [Example Query](https://duckduckgo.com/?q=css+background-position&ia=about)

  css background-position search

[See more related Instant Answers](https://duck.co/ia?repo=fathead)

## Get Started
- [x] 1) Claim this issue by commenting below
- [x] 2) Review our [Contributing Guide](https://github.com/duckduckgo/zeroclickinfo-fathead/blob/master/CONTRIBUTING.md)
- [x] 3) [Set up your development environment](https://docs.duckduckhack.com/welcome/setup-dev-environment.html), and fork this repository
- [x] 4) Create a new Instant Answer Page: https://duck.co/ia/new_ia (then let us know, here!)
- [ ] 5) Create the Fathead
- [ ] 6) Create a Pull Request
- [ ] 7) Ping @bfmags for a review

## Resources
- Join [DuckDuckHack Slack](https://quackslack.herokuapp.com/) to ask questions
- Join the [DuckDuckHack Forum](https://forum.duckduckhack.com/) to discuss project planning and Instant Answer metrics
- Read the [DuckDuckHack Documentation](https://docs.duckduckhack.com/) for technical help

### PR DESCRIPTION
**Pull request by adityatandon007**
**New Express Js Fathead**

## Description of new Instant Answer, or changes

A fathead for Express JS
source: http://expressjs.com/en/api.html

## Related Issues and Discussions
 
Fixes #811 

## People to notify

@pjhampton @moollaza 

## Testing & Review
To be completed by Language Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [x] Title follows correct format (Specifies Instant Answer + Purpose)
- [x] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [x] Instant Answer page is correctly filled out and contains:
    - One topic for the Search Space Language (Java, Python, Scala, Ruby, etc.)
    - One topic from: Reference, Help, Libraries, Tools
        - Documentation Fatheads are considered "Reference"
    - Description, Data source, and 2+ example queries
    - Perl Module (e.g. "DDG::Fathead::PerlDoc" -- we only need a name, not an actual file)
    - Source Name (for "More at <source_name>" link)
    - Source Domain (must contain http:// or https:// -- can be the same as Data Source)
    - Source Info (used as Subtitle for each Article -- usually matches the IA Name)
    - 'Skip Abstract' is checked off
    - Source ID (ping @moollaza to assign one, once Fathead is ready for Beta deploy)

**Code**
- [ ] Uniformly indented, well commented
- [ ] Fetch.sh and Parse.sh run without errors
- [ ] Output contains no blank lines, or multi-line entries
- [ ] Fathead Tests are passing (run `$ duckpan test <fathead_id>`)
    - Tester should report any failures

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/express_js

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by adityatandon007** [comment]
@mollaza @bfmags please review this

File: lib/fathead/express_js/downloads/docs.html [hunk 1]
```diff
@@ -0,0 +1,3311 @@
+<!DOCTYPE html>
```
Comment (moollaza) [inline lib/fathead/express_js/downloads/docs.html]: @adityatandon007 this should not be committed to the repo. Please remove

Comment (adityatandon007) [inline lib/fathead/express_js/downloads/docs.html]: @moollaza deleted the file buddy

**Comment by moollaza** [comment]
@adityatandon007 it doesn't look like your removal worked. I still see the downloads.html in the list of modified files\

**Comment by adityatandon007** [comment]
@moollaza I don't know what happened I don't see the `docs.html` in my local repository. I have deleted  and pushed the commit. I don't know why it is not showing in the list of modified files. Though in the commit section it shows the commit of _deleting docs.html_ 

screenshot from 2017-03-14 13-33-59

**Comment by adityatandon007** [comment]
@moollaza please review and merge this if it looks good. I want to see this live

**Comment by pjhampton** [comment]
Hey @adityatandon007!

[DuckDuckHack is now in Maintenance Mode](https://duckduckhack.com/) and from now on, we are only accepting issues and PRs for essential bugs and bug fixes.

Unfortunately, this is non-essential and will be closed.

We appreciate you taking the time to contribute and apologize for not being able to include your work.

**Pull request closed without merging**